nautobot/extras/signals.py

```diff
@@ -439,6 +440,143 @@ def dynamic_group_update_cached_members(sender, instance, **kwargs):
+            changes to the source data (at least those changes enacted via SQL).
+
+    This signal is responsible for creating or updating the triggers and functions. Each eligible table
+    (identified by ORM models) gets three triggers registered, one each for INSERT, UPDATE, and DELETE
+    statement event types. This is because both PostgreSQL and MySQL limit triggers to a single table and
+    action type pairing. Each trigger calls a single function that is centrally defined and implements the
+    business logic of updating the dirty flag on any group that references the trigger-sourced table as an
+    involved table.
+
+    The obvious downside to this approach is that we do not scope the trigger events to specific data since
+    all we can reasonably know about are the table names involved in a dynamic group’s queryset. This means
+    we fire the trigger for any events related to the table and look for relevant groups to update. At first
+    glance, this seems inefficient; however, because these are triggers run directly by the database, they
+    are, in fact, very fast, and the actual update query is simple. We can scope the triggers in PostgreSQL
+    to statement-based execution, but MySQL only offers row-based execution. Currently, this distinction is
+    irrelevant because Nautobot rarely modifies more than a single row per statement.
+    """
+    with connection.cursor() as cursor:
+        db_vendor = connection.vendor
+
+        # PostgreSQL specific trigger creation
+        if db_vendor == 'postgresql':
+            trigger_function_sql = """
+            CREATE OR REPLACE FUNCTION set_group_dirty_flag() RETURNS TRIGGER AS $$
+                BEGIN
+                    UPDATE extras_dynamicgroup
+                    SET _maybe_dirty = true
+                    WHERE _involved_tables @> to_jsonb(TG_TABLE_NAME)::jsonb;
+                    RETURN NULL;
+                END;
+            $$ LANGUAGE plpgsql;
+            """
+
+            trigger_sql_template = """
+            DROP TRIGGER IF EXISTS tr_i_group_dirty_{table_name} ON {table_name};
+            CREATE TRIGGER tr_i_group_dirty_{table_name}
+            AFTER INSERT ON {table_name}
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION set_group_dirty_flag();
+
+            DROP TRIGGER IF EXISTS tr_u_group_dirty_{table_name} ON {table_name};
+            CREATE TRIGGER tr_u_group_dirty_{table_name}
+            AFTER UPDATE ON {table_name}
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION set_group_dirty_flag();
+
+            DROP TRIGGER IF EXISTS tr_d_group_dirty_{table_name} ON {table_name};
+            CREATE TRIGGER tr_d_group_dirty_{table_name}
+            AFTER DELETE ON {table_name}
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION set_group_dirty_flag();
+            """
+
+        # MySQL specific trigger creation
+        elif db_vendor == 'mysql':
```

Comment:
Does this case cover Dolt as well?

Reply:
Good question! I have no idea, I think so though.